core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/GroupIdResolverCache.java

```diff
@@ -0,0 +1,237 @@
+/*
+ *   Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT     WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.wso2.carbon.user.core.common;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.xml.StringUtils;
+
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+
+/**
+ * Handle Cache in Group ID resolving flow.
+ */
+public class GroupIdResolverCache {
+
+    private static Log log = LogFactory.getLog(GroupIdResolverCache.class);
+    private static GroupIdResolverCache groupIdResolverCache = new GroupIdResolverCache();
+    private static final String GROUP_ID_RESOLVER_CACHE_MANAGER = "GROUP_ID_RESOLVER_CACHE_MANAGER";
+
+    private GroupIdResolverCache() {
+    }
+
+    /**
+     * Gets a new instance of GroupIdResolverCache.
+     *
+     * @return A new instance of GroupIdResolverCache.
+     */
+    public synchronized static GroupIdResolverCache getInstance() {
+
+        return groupIdResolverCache;
+    }
+
+    private Cache<String, Group> GroupIdResolverCache(String cacheName) {
+
+        Cache<String, Group> cache;
+        CacheManager cacheManager = Caching.getCacheManagerFactory().getCacheManager(GROUP_ID_RESOLVER_CACHE_MANAGER);
+        cache = cacheManager.getCache(cacheName);
+        return cache;
+    }
+
+    /**
+     * Add a cache entry.
+     *  @param key       Key which cache entry is indexed.
+     * @param entry     Actual object where cache entry is placed.
+     * @param cacheName Name of the cache.
+     * @param tenantId  Tenant ID.
+     */
+    public void addToCache(String key, Group entry, String cacheName, int tenantId) {
+
+        if (validateAddToCacheRequest(key, entry, cacheName)) return;
```

Comment:
Shall we use 'isValidCacheRequest' like method name, as this is a boolean operation.
When reading the method, it does not look right. It feels like if request is valid 'return'.
it should be like
if(isValid()){
}else{
returnl
}